qiskit/providers/backend.py

```diff
@@ -229,3 +240,435 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class QubitProperties:
```

Comment:
Just a suggestion. How about providing a comparison table between V1 and V2 like below somewhere after completing the design? I think it would help developers a lot.

Qubit properties provided by IBM backends are:
t1, t2, frequency, anharmonicity, readout_error, prob_meas0_prep1, prob_meas1_prep0, readout_length.

We can get those values in BackendV1 by
```
backendv1.properties().t1(0)
backendv1.properties().t2(0)
backendv1.properties().frequency(0)
backendv1.properties().qubits[0][3].value  # anharmonicity
backendv1.properties().readout_error(0)
backendv1.properties().qubits[0][5].value  # prob_meas0_prep1
backendv1.properties().qubits[0][6].value  # prob_meas1_prep0
backendv1.properties().readout_length(0)
# list index (e.g. [3]) access may fail
```
Using current BackendV2,
```
backendv2.qubit_properties(0).t1 (or backend.t1(0)?)
backendv2.qubit_properties(0).t2 (or backend.t2(0)?)
backendv2.qubit_properties(0).frequency
backendv2.qubit_properties(0).properties.anharmonicity ?
backendv2.target["measure"][(0,)].error
backendv2.qubit_properties(0).properties.prob_meas0_prep1 ?
backendv2.qubit_properties(0).properties.prob_meas1_prep0 ?
backendv2.target["measure"][(0,)].length
# attribute (e.g. `.anharmonicity`) access may fail
```

Reply:
I like this idea, I have a follow up after this PR merges to update the top level documentation in https://qiskit.org/documentation/apidoc/providers.html to make BackendV2 the default (mainly for the writing a provider section). But I will make a note to add a user migration guide section as part of that, and document these user changes going from v1 to a v2 backend.

Reply:
Yep for 1 that's expected pattern. In the IBM provider we'll make a `IBMQubitProperties` or `TransmonQubitProperties` subclass and expose an `anharmonicity` attribute at the same level.

For 2 I updated `InstructionProperties` to work the same way. For an example see: https://github.com/Qiskit/qiskit-terra/pull/5885/files#diff-a9905c867e5494f06b2e47d0a5be3d6eea2d9d7237ac548f6a2fa3bff9e5e0aeR845-R859

Reply:
I have the PR adding this documentation up here: https://github.com/Qiskit/qiskit-terra/pull/7323